app/code/core/Mage/Catalog/Helper/Product/View.php

```diff
@@ -58,6 +58,11 @@ public function initProductLayout($product, $controller)
 
         $update->addHandle('PRODUCT_TYPE_' . $product->getTypeId());
         $update->addHandle('PRODUCT_' . $product->getId());
+
+        if ($settings->getPageLayout()) {
+            $update->addHandle($settings->getPageLayout());
```

Comment:
I think you should actually be using this, no?
   
``$controller->getLayout()->helper('page/layout')->applyHandle($settings->getPageLayout());

Reply:
Actually, looking at the core config XML around `global/page/layouts` I see that the layout code and the layout handle *don't* match for the core page layouts. They use a `page_` prefix on all of the page layout handles. This is even more of a reason to apply the page layout handle via the helper.

Reply:
👍 I'll change that.

Reply:
Also, the `if` surrounding it isn't strictly needed. The `page/layout::applyHandle` code will be a no-op if the page layout code is invalid (null and '' are invalid).